### ISSUE TITLE
Freshwater budget

### ISSUE THREAD

**Issue by adele157** [pedrocol/basal_mom5-collaborative-project#68]
It would be great to repeat this budget analysis we did for the old simulations.

How does the freshwater budget, and its components, integrated over the shelf change? There will be changes due to the input salinity of the basal melt, changing salinity restoring, changing sea ice formation/melt, other air-sea FW fluxes (evap), and cross-slope transports of fresh/salty waters.

Old issue is [here](https://github.com/pedrocol/basal_mom5-collaborative-project/issues/36).

**Comment by fabiobdias** [comment]
Here is the surface freshwater budget components integrated over the shelf (ht < 1000m) in the last year (annual averaged):

![Screenshot 2023-08-24 at 1 37 25 pm](https://github.com/pedrocol/basal_mom5-collaborative-project/assets/8631257/46468dc3-2867-4c85-85da-d5884f576074)

Salt fluxes from sea-ice (which I believe is accounted in the "melt" term) and salinity restoration are converted to equivalent freshwater by:

FW_ice = 1000*sfc_salt_flux_ice/SSS
FW_restore = 1000*sfc_salt_flux_restore/SSS

The main differences are (1) reduction of the river/runoff term and (2) increase of the sea-ice salinification net effect in the Basal and ICB runs compared to the control.

The restoring term is smaller than other terms (by 2 orders of magnitude) but also changes across experiments:

![Screenshot 2023-08-24 at 1 47 31 pm](https://github.com/pedrocol/basal_mom5-collaborative-project/assets/8631257/1e828682-00f6-45ea-8a1b-2f053bbd1d13)

**Comment by fabiobdias** [comment]
And here some annual average maps:

1) net surface fw flux (pme_river):
![Screenshot 2023-08-24 at 2 45 50 pm](https://github.com/pedrocol/basal_mom5-collaborative-project/assets/8631257/e9a01138-b32b-4c75-8993-4c22b30bfdf1)

2) surface fw flux due to sea-ice (melt):
![Screenshot 2023-08-24 at 2 46 30 pm](https://github.com/pedrocol/basal_mom5-collaborative-project/assets/8631257/6fdc8c02-a6d4-47e3-9a00-82e229e630d2)

**Comment by fabiobdias** [comment]
Some maps for the salt flux converted to FW (note the different colorbar limits in comparison with previous post):

3. salt flux due to sea ice converted to FWF:
![Screenshot 2023-08-24 at 3 09 12 pm](https://github.com/pedrocol/basal_mom5-collaborative-project/assets/8631257/5ea38900-1d89-468b-836b-577d1a4ecca6)

4. same but due to salinity restoring:
![Screenshot 2023-08-24 at 3 43 00 pm](https://github.com/pedrocol/basal_mom5-collaborative-project/assets/8631257/d9b31242-7464-43e6-8412-7d957d840853)

**Comment by adele-morrison** [comment]
I'm not sure I understand what's going on here @fabiobdias. Did you include the new freshwater input from the basal and iceberg terms? These aren't included in `river` by the looks of the above plot. What are the diagnostic names to include the freshwater from basal and icebergs @pedrocol? 

Also is `melt` = `wfiform` + `wfimelt`? The `melt` diagnostic is incorrect.

**Comment by pedrocol** [comment]
The variable names are:
- for basal, is basal_fwflx (3d) and basal_fwflx2d (2d, vertically integrated)
- for icb, is icb_fwflx (3d) and icb_fwflx2d (2d, vertically integrated)

**Comment by fabiobdias** [comment]
No, I haven’t taken those terms into account – thanks Pedro. I will re-do these and double-check the melt term.

From: Pedro ***@***.***>
Date: Thursday, 2 November 2023 at 10:31 am
To: pedrocol/basal_mom5-collaborative-project ***@***.***>
Cc: Fabio Boeira Dias ***@***.***>, Mention ***@***.***>
Subject: Re: [pedrocol/basal_mom5-collaborative-project] Freshwater budget (Issue #68)

The variable names are:

  *   for basal, is basal_fwflx (3d) and basal_fwflx2d (2d, vertically integrated)
  *   for icb, is icb_fwflx (3d) and icb_fwflx2d (2d, vertically integrated)

**Comment by aekiss** [comment]
> Salt fluxes from sea-ice (which I believe is accounted in the "melt" term) and salinity restoration are converted to equivalent freshwater by:

> FW_ice = 1000 x sfc_salt_flux_ice/SSS
FW_restore = 1000 x sfc_salt_flux_restore/SSS

I think there are 2 problems here:

1. the FW flux associated with sea ice melt should be calculated using the fixed bulk salinity of sea ice (`ice_salt_concentration` = 0.005 kg salt / kg ice), not SSS

2. Is the salinity restoring done as a salt flux (`salt_restore_as_salt_flux` = true)? If so I don't think it makes sense to convert salinity restoring to a FW flux, as there is no associated FW change.

See sec 3.5.2, 3.5.3 of the [draft tech report](https://github.com/COSIMA/ACCESS-OM2-1-025-010deg-report).

**Comment by aekiss** [comment]
To possibly clarify point 2: the salinity changes under restoring are purely due to salt mass flux, not FW mass flux.
You can only form a budget with extensive variables like salt mass and water mass, not intensive variables like salinity.

**Comment by fabiobdias** [comment]
Updated surface freshwater budget figure. Some notes:

- melt = wfiform + wfimelt
- pme_river = pme_net + river
- pme_net includes the melt term (so I plotted [pme_net - melt])
- 3rd column includes:
  - river (control)
  - river + basal_fwflx2d (Basal)
  - river +  basal_fwflx2d + icb_fwflx2s (ICB)
  so I plotted only the extra terms in the latest column for sanity check.

Also, I'm still including the converted FWF from sfc_salt_flux_ice and sfc_salt_flux_restore terms, but following Andrew comments above we should remove the restore from the final figure. Is the sfc_salt_flux_ice already included in the wfiform/wfimelt terms as well? If so, we could remove that too.

<img width="817" alt="Screenshot 2024-02-09 at 8 01 12 pm" src="https://github.com/pedrocol/basal_mom5-collaborative-project/assets/8631257/56a80406-13c5-4747-a59a-e5b52f8d3ee9">

**Comment by fabiobdias** [comment]
Just want to add a note here as I got a bit confused when closing the budget. The `pme_net` and `pme_river` terms do NOT include the additional basal/iceberg fluxes (`basal_fwflx2d` and `icb_fwflx2d`, in the same way they aren't included in the `river` output). This changes the values of the net FWF (pme_river in the plot above). New corrected plot below.

Recapping, we have:

`NET = pme_river + basal_fwflx2d + icb_fwflx2d`
`MELT = wfiform + wfimelt`
`PME = pme_net - MELT`
`RIVER = river + basal_fwflx2d + icb_fwflx2d`

considering SSSi = 1/0.005 as the "fixed bulk salinity of sea ice (ice_salt_concentration = 0.005 kg salt / kg ice)", and SSS the model surface salinity:

`FW_ICE = -sfc_salt_flux_ice*1000)/SSSi`
`FW_RESTORE = -(sfc_salt_flux_restore*1000)/SSS`

The surface FW budget are given by:

`NET = PME + RIVER + MELT` (so leaving FW_ICE/RESTORE out of the FW budget as Andrew suggested).

In summary, for models using this depth distributed basal fluxes and/or iceberg fluxes, the `basal_fwflx2d` and `icb_fwflx2d` needs to be summed to the `pme_net`, `pme_river` and `river` diagnostics. Not sure if we should do this internally in MOM5 to be consistent?

New fig below - I'll be adding this and the SHF budget in the overleaf next:

<img width="833" alt="Screenshot 2024-02-27 at 6 38 55 pm" src="https://github.com/pedrocol/basal_mom5-collaborative-project/assets/8631257/07e52f37-973b-43ec-8866-da2bc45137d8">

**Comment by pedrocol** [comment]
Hi Fabio, I'm not sure I'm following, the code accounts for basal and icb fluxes for the zero_net_water_coupler (line 4081 in /home/552/pc5520/access-om2/src/mom/src/mom5/ocean_core/ocean_sbc.F90)

pme_river(i,j) = pme(i,j) + river(i,j) + basal(i,j) + icb(i,j) - melt(i,j) - wrk1_2d(i,j)

In this way, when splitting runoff (river) in basal+icb, the rest of the fields remain unchanged.

I did a mass balance and it was ok, would you mind sharing your jupyter notebook?

**Comment by fabiobdias** [comment]
Hmm... now I got confused. Are you saying the `basal_fwflx2d/icb_fwflx2d` aren't included in the `river` but they are in `pme_river`?

Following my analyses, I think it's either not included in both (as my note above), or it's already included in both. I believe it isn't included in both as `river` in the control only matches with the total runoff from Basal_gade and NoBasal_gade experiments if I add the `basal_fwflx2d` to `river`. That aligns with Adele's suggestion from above:

> I'm not sure I understand what's going on here @fabiobdias. Did you include the new freshwater input from the basal and iceberg terms? These aren't included in `river` by the looks of the above plot. What are the diagnostic names to include the freshwater from basal and icebergs @pedrocol?

If I added `basal_fwflx2d` (and `icb_fwflx2d`) to `river` but don't do it for `pme_river`, I can't close the budget. 

`river` in the ICB expt is zero, so `basal_fwflx2d+icb_fwflx2d` accounts for all runoff (which is less than the total runoff in other experiments and that contributes to a larger difference in the NET FWF relative to control compared to Basal_gade). I'm not sure this is something we should expect.

I'm going to do a pull with the fw budget script. I have also a spreadsheet with the shelf-integrated values that I can share if you want (these values are all printed in the notebook).

**Comment by pedrocol** [comment]
basal and icb arrays are not included in river:

river(i,j) = runoff(i,j) + calving(i,j); where calving = 0.0 and runoff(i,j) = 0.0, but runoff was previously split in basal(i,j) and icb(i,j). So, river(i,j) + basal(i,j) + icb(i,j) remain unchanged, as well as pme_river. 

Indeed, river is expected to be zero in the ICB experiment.

I think the problem comes from the calculation region, you are calculating the quantities on-shelf only, and the ICB field is also present further away.

**Comment by fabiobdias** [comment]
I've just re-done this but integrated over the whole domain (up to 60S or so), but I still think `pme_river` doesn't include `basal_fwflx2d+icb_fwflx2d `fluxes. 

For the whole domain (S.O.), the imbalance in the NET - (PME+RIVER+SEA-ICE) is the same. I still need to add `basal_fwflx2d+icb_fwflx2d` to close the surface FW budget (see table and figs below).

For the ICB experiment, what you said holds when I integrated for the whole SO (not only cont. shelf)–the ICB total river fluxes (`river+basal_fwflx2d+icb_fwflx2d`) get pretty close to the river in the control run (or `river+basal_fwflx2d `in the Basal_gade expt). 

The integrated values are:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/f.obeira_dias/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/f.obeira_dias/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">


  | pme_river | pme_net - melt | river + basal + icb |   | melt |   |  
-- | -- | -- | -- | -- | -- | -- | --
1E8 (kg s-1) | NET | PME | RUNOFF | basal/icb 2dflx | SEA-ICE | pme_net +river + basal + icb | minus (basal+icb)
CTRL | 3.55587328 | 1.36628928 | 0.87460016 | 0 | 1.31498392 | 3.55587336 |  
BG | 3.04362976 | 1.35479456 | 0.87459888 | 0.45816592 | 1.27240208 | 3.50179552 | 3.0436296
ICB | 2.60107904 | 1.34355968 | 0.87818384 | 0.87818384 | 1.25751928 | 3.4792628 | 2.60107896



</body>

</html>


Figures below show (top) NET as pme_river and (bottom) NET as [pme_river + basal2dflx + icb2dflx].

<img width="706" alt="Screenshot 2024-02-29 at 8 01 36 pm" src="https://github.com/pedrocol/basal_mom5-collaborative-project/assets/8631257/942ab560-96f2-479c-baee-9c4b022e9e14">


<img width="686" alt="Screenshot 2024-02-29 at 8 02 03 pm" src="https://github.com/pedrocol/basal_mom5-collaborative-project/assets/8631257/e4674653-3c3b-47df-a4b0-d83677898e6c">

I see the code snip that you mentioned above, but I think something isn't right still. There is any way the `pme_river` output diagnostic (that I'm using) and the internal `pme_river` (from the code you showed) variables aren't equal?

**Comment by pedrocol** [comment]
Yes, you need to add basal and icb to pme in order to close the budget. This is because the runoff was separated into basal and icb. If I got it correctly the difference between Control and Basal is 3.55587336 vs 3.50179552 (~1.5%), right?

I saw you had 8 terms before and 7 now, would this be the cause?

The very small difference in the icb case (3.50179552 vs 3.4792628, ~0.6%) comes from the rescaling we did to the icb flux, since the NEMO land mask was different to the access-om2 mask.

**Comment by fabiobdias** [comment]
> Yes, you need to add basal and icb to pme in order to close the budget. This is because the runoff was separated into basal and icb. If I got it correctly the difference between Control and Basal is 3.55587336 vs 3.50179552 (~1.5%), right?
NET in Control is 3.55587328, Basal_gade is 3.50179552 (-1.5%) and ICB is 3.4792628 (-2.1%).

> 
> I saw you had 8 terms before and 7 now, would this be the cause?

I just removed the salt flux due to sea-ice converted to FWF (sfc_salt_flux_ice) from the previous plot.
> 
> The very small difference in the icb case (3.50179552 vs 3.4792628, ~0.6%) comes from the rescaling we did to the icb flux, since the NEMO land mask was different to the access-om2 mask.
Got it.

**Comment by fabiobdias** [comment]
So in summary the surface freshwater budget here is obtained by:

`pme_river + basal2dflx + icb2dflx = pme_net + river + basal2dflx + icb2dflx
`
and using the nomenclature used in the figures above we have:

```
NET = pme_river + basal2dflx + icb2dflx
PME = pme_net - SEA-ICE
SEA-ICE = wfiform + wfimelt
SEA-ICE FORM = wfiform
SEA-ICE MELT = wfimelt
RIVER = river + basal2dflx + icb2dflx
```

So

`NET = PME + RIVER + SEA-ICE`

The equivalent FWF from salinity restoring will be kept in the figure for comparison proposes and it didn't contribute to the FW budget.

**Comment by fabiobdias** [comment]
Final figure and key points:

![SFWF_components_annual_hint_shelf_yr10_FinalFig_WO_noBG](https://github.com/pedrocol/basal_mom5-collaborative-project/assets/8631257/9d28240e-9213-4da0-8630-b143c2699289)

- Negative NET SFW *over the shelf* increases in both Basal_gade and ICB runs relative to control (by 21.3 and 40.3% respectively)
- This is dominated by sea-ice changes, where sea-ice formation increases by about 17/18.9% (BG/ICB), offset by an increase of 27.8/29.5 % (BG/ICB) in sea-ice melting; total Sea-ice contribution changes is 11.9/13.9% increase in BG/ICB;
- Runoff/river decreases in ICB (compared to CTRL) due to the icb2dflx that is distributed further north of the shelf break (reduction of 19.8% relative to CTRL), therefore contributing to enhanced Negative NET SFW in ICB expt;
- PME and salinity restoring have a negligible contribution compared to other contributors to the SFC FW Budget.

**Comment by fabiobdias** [comment]
I've updated the fig above with the new runs but something looks weird with the BasalGadeBrine PME (pme_net and pme_river fields)... and I couldn't think of any reason PME should increases on that run:

![Screenshot 2024-08-21 at 3 35 35 PM](https://github.com/user-attachments/assets/62697ab1-d5fc-4e6e-872e-4b0403f8ad79)

**Comment by pedrocol** [comment]
Thanks for the plot Fabio, pme increases since no longer contains brine (<0, wfiform). You need to include "briner_fwflx2d" in pme in order to compare to the other two simulations

**Comment by fabiobdias** [comment]
Ah great, thanks @pedrocol 
There is anything equivalent for the heat fluxes that I should also account for?

**Comment by pedrocol** [comment]
No, that would be it.
I just saw there's a difference in the runoff amount, I'll check that.

**Comment by fabiobdias** [comment]
Thanks Pedro, here's the update figured:

![SFWF_components_annual_hint_shelf_yr10_FinalFinalFig](https://github.com/user-attachments/assets/dfaa49f8-d1e4-4b1b-b8ee-f7e99e0a86e5)

So now I've accounted the briner_fwflx2d into the Net FWF and PME terms. There's still this weird difference in the runoff. Maybe that's trying to compesate the Brine negative fluxes?

**Comment by pedrocol** [comment]
Thanks Fabio, the Brine negative flux equals sea-ice form., so that's ok, PME looks fine as well. The runoff value seems ok for Control, but it should be about half of it for the three Basal runs.

**Comment by fabiobdias** [comment]
I'm posting here the time series over the 10 years simulations for individual surface heat and freshwater budgets @AndyHoggANU 

I'm not sure these add much on the discussion we already have on the m/s but happy to hear other ppl thoughts.

- Freshwater budget:

![Image](https://github.com/user-attachments/assets/75064789-620a-4823-99f8-a9f7a1de5d9c)


Heat budget:

![Image](https://github.com/user-attachments/assets/694d9b72-860e-46c1-9ff0-44206259bcb1)

![Image](https://github.com/user-attachments/assets/21c52668-1991-452b-9cdc-4c3a4b8f4975)

**Comment by fabiobdias** [comment]
I've give a closer look to the seasonal freshwater and heat budgets integrated over the shelf. This time I'm plotting anomalies referent to the control simulation. 

So this is the FW budget that we currently have on the m/s (annual avg over the last 3 years):

<img width="880" height="1000" alt="Image" src="https://github.com/user-attachments/assets/2d18fb33-8d3c-4131-add6-f234a706598d" />

and these are the time series with seasonal cycle:

<img width="1033" height="1560" alt="Image" src="https://github.com/user-attachments/assets/8d2fec4c-ee77-472a-acab-dd2c83b0c5dd" />

The time series show better the differences (i.e. larger anomalies from sea-ice processes in winter/summer, specially in basal_LH), but the message remains the same (that sea-ice dominate changes in the FW budget).
